### PR TITLE
feat: add possibility to store/retrieve multiple values at the same time

### PR DESCRIPTION
**Pull request by GuillaumeOnepilot**
Implement #41 

I'm really open to discussion because I don't think this feature should be on the same method as `set`/`get` since it make the thing confusing, but I tried to follow the chrome API.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -369,13 +369,49 @@ export type StorageOptions = ConstructorParameters<typeof BaseStorage>[0]
...
+          const nsKey = this.getNamespacedKey(index);
+          const jsonValue = JSON.stringify(value);
+          return this.rawSet(nsKey, jsonValue);
+        })
+      );
+
+      return null;
+    }
+
+    if (!rawValue) {
```
Comment (louisgv) [inline src/index.ts:411]: this doesn't check for strict null, thus an empty string would throw the error as well.

**Comment by louisgv** [comment]
@GuillaumeOnepilot Thanks for the PR and sorry for the late reply - I've been away for a bit. I don't think we have to worry about following the chrome API. We should make it easy to use and obvious for the API user. IMO, coupling the twos seems weird, but could make sense. Can we also provide a `getItems` and `setItems`, and maybe the `get` and `set` method would be the one where you can do either?

Would be a slight refactor, but I think it'd make it nicer.

**Comment by GuillaumeOnepilot** [comment]
@louisgv Thanks for the answer, I agree coupling is following the chrome API, but I find it hard to use and hard to type properly (I did not manage to achieve proper typing of the method depending on the parameters), so I will try to that as soon as possible.

File: src/index.test.ts [hunk 2]
```diff
@@ -411,4 +455,4 @@ describe("Storage - Basic CRUD operations with namespace", () => {
       `${namespace}${testKey2}`
     )
   })
-})
+})
```
[the remark this answers is not in the document]
Comment (GuillaumeOnepilot) [reply src/index.test.ts]: It's done :)

**Comment by GuillaumeOnepilot** [comment]
Hey @louisgv, 

Any news ?

**Comment by GuillaumeOnepilot** [comment]
@louisgv any updates on this matter?

**Comment by sleekslush** [comment]
@GuillaumeOnepilot and @louisgv I was thinking about this as well, but approached it in a different way. Biggest difference is I just use the bulk syntax in chrome.storage and browser.storage instead of doing a bunch of async calls for each operation. Let me know your thoughts https://github.com/PlasmoHQ/storage/pull/51

**Comment by louisgv** [comment]
@GuillaumeOnepilot argh I don't check GitHub's inbox notification as often 🙈 Looking now

**Comment by louisgv** [comment]
I think #51 likely supersedes this since the array argument works in firefox: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/get

BTW the best way to get a hold of me is likely Discord. I have GH notification on my phone but it's swarmed by other repos xd....

**Comment by GuillaumeOnepilot** [comment]
@louisgv No issue I will close this one then

**Pull request closed without merging**